src/components/GraphiQL.js

```diff
@@ -694,6 +702,18 @@ export class GraphiQL extends React.Component {
     editor.setValue(print(parse(editor.getValue())));
   };
 
+  handleMergeQuery = () => {
+    const editor = this.getQueryEditor();
+    const query = editor.getValue();
+
+    if (!query) {
+      return;
+    }
+
+    const ast = parse(query);
```

Comment:
This will throw if the query is invalid - please handle this case by `try/catch`ing and consider implementing the suggestion above :)